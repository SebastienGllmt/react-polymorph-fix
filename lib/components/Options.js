'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Options = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _withTheme = require('./HOC/withTheme');

var _themes = require('../utils/themes');

var _props6 = require('../utils/props');

var _ = require('.');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// internal utility functions


// import constants


var OptionsBase = function (_Component) {
  _inherits(OptionsBase, _Component);

  // TODO: Does this get used? Don't think so.

  // define static properties
  function OptionsBase(props) {
    _classCallCheck(this, OptionsBase);

    var _this = _possibleConstructorReturn(this, (OptionsBase.__proto__ || Object.getPrototypeOf(OptionsBase)).call(this, props));

    _initialiseProps.call(_this);

    var context = props.context,
        themeId = props.themeId,
        theme = props.theme,
        themeOverrides = props.themeOverrides;


    _this.state = {
      composedTheme: (0, _themes.composeTheme)((0, _themes.addThemeId)(theme || context.theme, themeId), (0, _themes.addThemeId)(themeOverrides, themeId), context.ROOT_THEME_API),
      highlightedOptionIndex: 0
    };
    return _this;
  }
  // declare ref types


  _createClass(OptionsBase, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      if (this.props.isOpen) {
        document.addEventListener('keydown', this._handleKeyDown, false);
      }
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      if (!this.props.isOpen && nextProps.isOpen) {
        document.addEventListener('keydown', this._handleKeyDown, false);
      } else if (this.props.isOpen && !nextProps.isOpen) {
        document.removeEventListener('keydown', this._handleKeyDown, false);
      }
      (0, _themes.didThemePropsChange)(this.props, nextProps, this.setState.bind(this));
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      document.removeEventListener('keydown', this._handleKeyDown, false);
    }

    // returns an object containing props, theme, and method handlers
    // associated with rendering this.props.options, the user can call
    // this in the body of the renderOptions function


    // ========= PRIVATE HELPERS =========

    // this needs to get passed to OptionsSkin and attached to each Option Li

  }, {
    key: 'render',
    value: function render() {
      // destructuring props ensures only the "...rest" get passed down
      var _props = this.props,
          skin = _props.skin,
          targetRef = _props.targetRef,
          theme = _props.theme,
          themeOverrides = _props.themeOverrides,
          onChange = _props.onChange,
          context = _props.context,
          optionsRef = _props.optionsRef,
          isOpen = _props.isOpen,
          rest = _objectWithoutProperties(_props, ['skin', 'targetRef', 'theme', 'themeOverrides', 'onChange', 'context', 'optionsRef', 'isOpen']);

      var _state = this.state,
          composedTheme = _state.composedTheme,
          highlightedOptionIndex = _state.highlightedOptionIndex;


      var OptionsSkin = skin || context.skins[_.IDENTIFIERS.OPTIONS];

      return _react2.default.createElement(OptionsSkin, _extends({
        getHighlightedOptionIndex: this.getHighlightedOptionIndex,
        getOptionProps: this.getOptionProps,
        handleClickOnOption: this.handleClickOnOption,
        highlightedOptionIndex: highlightedOptionIndex,
        isHighlightedOption: this.isHighlightedOption,
        isOpen: isOpen,
        isSelectedOption: this.isSelectedOption,
        optionsRef: optionsRef,
        setHighlightedOptionIndex: this.setHighlightedOptionIndex,
        targetRef: targetRef,
        theme: composedTheme
      }, rest));
    }
  }]);

  return OptionsBase;
}(_react.Component);

OptionsBase.displayName = 'Options';
OptionsBase.defaultProps = {
  context: (0, _withTheme.createEmptyContext)(),
  isOpen: false,
  isOpeningUpward: false,
  noResultsMessage: 'No results',
  options: [],
  resetOnClose: false,
  theme: null,
  themeId: _.IDENTIFIERS.OPTIONS,
  themeOverrides: {},
  toggleOpen: function toggleOpen() {}
};

var _initialiseProps = function _initialiseProps() {
  var _this2 = this;

  this.close = function () {
    var _props2 = _this2.props,
        isOpen = _props2.isOpen,
        onClose = _props2.onClose,
        resetOnClose = _props2.resetOnClose,
        toggleOpen = _props2.toggleOpen;

    if (isOpen && toggleOpen) toggleOpen();
    _this2.setState({
      highlightedOptionIndex: resetOnClose ? 0 : _this2.state.highlightedOptionIndex
    });
    if (onClose) onClose();
  };

  this.getHighlightedOptionIndex = function () {
    // If nothing is higlighted, highlight selected option
    // In case nothing is selected, highlight first option
    var _props3 = _this2.props,
        options = _props3.options,
        isOpeningUpward = _props3.isOpeningUpward;

    var currentIndex = _this2.state.highlightedOptionIndex;
    var index = 0;

    if (currentIndex !== null) {
      index = currentIndex;
    }

    if (isOpeningUpward) return options.length - 1 - index;
    return index;
  };

  this.setHighlightedOptionIndex = function (optionIndex) {
    if (!_this2.isHighlightedOption(optionIndex) && _this2.isDisabledOption(optionIndex)) {
      _this2.setState({ highlightedOptionIndex: optionIndex });
    }
  };

  this.isSelectedOption = function (optionIndex) {
    var _props4 = _this2.props,
        options = _props4.options,
        isOpeningUpward = _props4.isOpeningUpward;

    var index = isOpeningUpward ? options.length - 1 - optionIndex : optionIndex;
    var option = options[index];
    return option && _this2.props.selectedOption === option;
  };

  this.isHighlightedOption = function (optionIndex) {
    return _this2.state.highlightedOptionIndex === optionIndex;
  };

  this.isDisabledOption = function (optionIndex) {
    var options = _this2.props.options;

    var option = options[optionIndex];
    return option && !option.isDisabled;
  };

  this.handleClickOnOption = function (option, event) {
    if (option) {
      if (option.isDisabled) return;
      if (_this2.props.onChange) _this2.props.onChange(option, event);
    }
    if (_this2.props.onBlur) _this2.props.onBlur(event);
    _this2.close();
  };

  this.getOptionProps = function () {
    var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

    var _onClick = _ref.onClick,
        _onMouseEnter = _ref.onMouseEnter,
        rest = _objectWithoutProperties(_ref, ['onClick', 'onMouseEnter']);

    var _props5 = _this2.props,
        isOpen = _props5.isOpen,
        themeId = _props5.themeId,
        options = _props5.options,
        selectedOptions = _props5.selectedOptions;
    var composedTheme = _this2.state.composedTheme;
    var isHighlightedOption = _this2.isHighlightedOption,
        isDisabledOption = _this2.isDisabledOption,
        handleClickOnOption = _this2.handleClickOnOption,
        setHighlightedOptionIndex = _this2.setHighlightedOptionIndex;


    return _extends({
      options: options,
      selectedOptions: selectedOptions,
      isOpen: isOpen,
      isHighlightedOption: isHighlightedOption,
      isDisabledOption: isDisabledOption,
      theme: composedTheme[themeId],
      onClick: function onClick(option, event
      // the user's custom onClick event handler is composed with
      // the internal functionality of Options (this.handleClickOnOption)
      ) {
        return (0, _props6.composeFunctions)(_onClick, handleClickOnOption)(option, event);
      },
      onMouseEnter: function onMouseEnter(index, event
      // user's custom onMouseEnter is composed with this.setHighlightedOptionIndex
      ) {
        return (0, _props6.composeFunctions)(_onMouseEnter, setHighlightedOptionIndex)(index, event);
      }
    }, rest);
  };

  this._handleSelectionOnKeyDown = function (event) {
    var options = _this2.props.options;

    if (options.length) {
      var _isOpeningUpward = _this2.props.isOpeningUpward;

      var currentIndex = _this2.state.highlightedOptionIndex;
      var reverseIndex = options.length - 1 - currentIndex;
      var highlightedOption = options[_isOpeningUpward ? reverseIndex : currentIndex];
      _this2.handleClickOnOption(highlightedOption, event);
    } else {
      event.preventDefault();
    }
  };

  this._handleHighlightMove = function (currentIndex, direction) {
    var options = _this2.props.options;

    if (options.length) {
      var lowerIndexBound = 0;
      var upperIndexBound = options.length - 1;
      var newIndex = direction === 'up' ? currentIndex - 1 : currentIndex + 1;

      // Make sure new index is within options bounds
      newIndex = Math.max(lowerIndexBound, Math.min(newIndex, upperIndexBound));

      if (options[newIndex].isDisabled) {
        // Try to jump over disabled options
        var canMoveUp = newIndex > lowerIndexBound;
        var canMoveDown = newIndex < upperIndexBound;
        if (direction === 'up' && canMoveUp || direction === 'down' && canMoveDown) {
          _this2._handleHighlightMove(newIndex, direction);
        }
      } else {
        _this2.setHighlightedOptionIndex(newIndex);
      }
    }
  };

  this._handleKeyDown = function (event) {
    var highlightOptionIndex = _this2.state.highlightedOptionIndex;
    switch (event.keyCode) {
      case 9:
        // Tab key: selects currently highlighted option
        event.preventDefault();
        _this2._handleSelectionOnKeyDown(event);
        break;
      case 13:
        // Enter key: selects currently highlighted option
        event.preventDefault();
        _this2._handleSelectionOnKeyDown(event);
        break;
      case 32:
        // Space key: selects currently highlighted option
        event.preventDefault();
        _this2._handleSelectionOnKeyDown(event);
        break;
      case 27:
        // Escape key: closes options if open
        _this2.close();
        break;
      case 38:
        // Up Arrow key: moves highlighted selection 'up' 1 index
        event.preventDefault(); // prevent caret move
        _this2._handleHighlightMove(highlightOptionIndex, 'up');
        break;
      case 40:
        // Down Arrow key: moves highlighted selection 'down' 1 index
        event.preventDefault(); // prevent caret move
        _this2._handleHighlightMove(highlightOptionIndex, 'down');
        break;
      default:
        _this2.props.resetOnClose && _this2.setHighlightedOptionIndex(0);
    }
  };
};

var Options = exports.Options = (0, _withTheme.withTheme)(OptionsBase);
//# sourceMappingURL=Options.js.map