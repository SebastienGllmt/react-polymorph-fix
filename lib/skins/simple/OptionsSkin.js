'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OptionsSkin = undefined;

var _isObject2 = require('lodash/isObject');

var _isObject3 = _interopRequireDefault(_isObject2);

var _isFunction2 = require('lodash/isFunction');

var _isFunction3 = _interopRequireDefault(_isFunction2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Bubble = require('../../components/Bubble');

var _BubbleSkin = require('./BubbleSkin');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// components
var OptionsSkin = exports.OptionsSkin = function OptionsSkin(props) {
  var getOptionProps = props.getOptionProps,
      getHighlightedOptionIndex = props.getHighlightedOptionIndex,
      handleClickOnOption = props.handleClickOnOption,
      isHighlightedOption = props.isHighlightedOption,
      isOpen = props.isOpen,
      isOpeningUpward = props.isOpeningUpward,
      isSelectedOption = props.isSelectedOption,
      noResults = props.noResults,
      noResultsMessage = props.noResultsMessage,
      optionsMaxHeight = props.optionsMaxHeight,
      optionRenderer = props.optionRenderer,
      options = props.options,
      optionsRef = props.optionsRef,
      render = props.render,
      setHighlightedOptionIndex = props.setHighlightedOptionIndex,
      targetRef = props.targetRef,
      theme = props.theme,
      themeId = props.themeId,
      toggleMouseLocation = props.toggleMouseLocation;


  var highlightedOptionIndex = getHighlightedOptionIndex();
  var isFirstOptionHighlighted = highlightedOptionIndex === 0;
  var sortedOptions = isOpeningUpward ? options.slice().reverse() : options;

  var renderOptions = function renderOptions() {
    // check for user's custom render function
    // if Options is being rendered via Autocomplete,
    // the value of props.render is renderOptions passed down from AutocompleteSkin
    if (!noResults && render) {
      // call user's custom render function
      return render(getOptionProps);
    } else if (!noResults && !render) {
      // render default simple skin
      return sortedOptions.map(function (option, index) {
        // set reference of event handlers in memory to prevent excess re-renders
        var boundSetHighlightedOptionIndex = setHighlightedOptionIndex.bind(null, index);
        var boundHandleClickOnOption = handleClickOnOption.bind(null, option);

        return _react2.default.createElement(
          'li',
          {
            role: 'presentation',
            'aria-hidden': true,
            key: index,
            className: (0, _classnames2.default)([theme[themeId].option, isHighlightedOption(index) ? theme[themeId].highlightedOption : null, isSelectedOption(index) ? theme[themeId].selectedOption : null, option.isDisabled ? theme[themeId].disabledOption : null]),
            onClick: boundHandleClickOnOption,
            onMouseEnter: boundSetHighlightedOptionIndex
          },
          renderOption(option)
        );
      });
    }
    // render no results message
    return _react2.default.createElement(
      'li',
      { className: theme[themeId].option },
      noResultsMessage
    );
  };

  var renderOption = function renderOption(option) {
    // check if user has passed render prop "optionRenderer"
    if (optionRenderer && (0, _isFunction3.default)(optionRenderer)) {
      // call user's custom rendering logic
      return optionRenderer(option);
    } else if ((0, _isObject3.default)(option)) {
      return _react2.default.createElement(
        'span',
        { className: theme[themeId].label },
        option.label
      );
    }
    return option;
  };

  // Enforce max height of options dropdown if necessary
  var optionsStyle = optionsMaxHeight == null ? null : {
    maxHeight: optionsMaxHeight + 'px'
  };

  return _react2.default.createElement(
    _Bubble.Bubble,
    {
      className: (0, _classnames2.default)([theme[themeId].options, isOpen ? theme[themeId].isOpen : null, isOpeningUpward ? theme[themeId].openUpward : null, isFirstOptionHighlighted && !noResults ? theme[themeId].firstOptionHighlighted : null]),
      isTransparent: false,
      skin: _BubbleSkin.BubbleSkin,
      isOpeningUpward: isOpeningUpward,
      isHidden: !isOpen,
      isFloating: true,
      targetRef: targetRef
    },
    _react2.default.createElement(
      'ul',
      {
        style: optionsStyle,
        ref: optionsRef,
        className: theme[themeId].ul,
        onMouseEnter: toggleMouseLocation,
        onMouseLeave: toggleMouseLocation
      },
      renderOptions()
    )
  );
};

// skins


// external libraries
//# sourceMappingURL=OptionsSkin.js.map