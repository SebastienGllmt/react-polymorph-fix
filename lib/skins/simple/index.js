'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleSkins = undefined;

var _SimpleSkins;

var _components = require('../../components');

var _AutocompleteSkin = require('./AutocompleteSkin');

var _BubbleSkin = require('./BubbleSkin');

var _ButtonSkin = require('./ButtonSkin');

var _CheckboxSkin = require('./CheckboxSkin');

var _FormFieldSkin = require('./FormFieldSkin');

var _HeaderSkin = require('./HeaderSkin');

var _InfiniteScrollSkin = require('./InfiniteScrollSkin');

var _InputSkin = require('./InputSkin');

var _LoadingSpinnerSkin = require('./LoadingSpinnerSkin');

var _ModalSkin = require('./ModalSkin');

var _OptionsSkin = require('./OptionsSkin');

var _ProgressBarSkin = require('./ProgressBarSkin');

var _RadioSkin = require('./RadioSkin');

var _SelectSkin = require('./SelectSkin');

var _SwitchSkin = require('./SwitchSkin');

var _TextAreaSkin = require('./TextAreaSkin');

var _TogglerSkin = require('./TogglerSkin');

var _TooltipSkin = require('./TooltipSkin');

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SimpleSkins = exports.SimpleSkins = (_SimpleSkins = {}, _defineProperty(_SimpleSkins, _components.IDENTIFIERS.AUTOCOMPLETE, _AutocompleteSkin.AutocompleteSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUBBLE, _BubbleSkin.BubbleSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUTTON, _ButtonSkin.ButtonSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.CHECKBOX, _CheckboxSkin.CheckboxSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.FORM_FIELD, _FormFieldSkin.FormFieldSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.HEADER, _HeaderSkin.HeaderSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INFINITE_SCROLL, _InfiniteScrollSkin.InfiniteScrollSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INPUT, _InputSkin.InputSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.LOADING_SPINNER, _LoadingSpinnerSkin.LoadingSpinnerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.MODAL, _ModalSkin.ModalSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.OPTIONS, _OptionsSkin.OptionsSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.PROGRESS_BAR, _ProgressBarSkin.ProgressBarSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.RADIO, _RadioSkin.RadioSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SELECT, _SelectSkin.SelectSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SWITCH, _SwitchSkin.SwitchSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TEXT_AREA, _TextAreaSkin.TextAreaSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOGGLER, _TogglerSkin.TogglerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOOLTIP, _TooltipSkin.TooltipSkin), _SimpleSkins);
//# sourceMappingURL=index.js.map