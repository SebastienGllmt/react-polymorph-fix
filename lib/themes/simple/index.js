'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleTheme = undefined;

var _SimpleTheme;

var _components = require('../../components');

var _SimpleAutocomplete = require('./SimpleAutocomplete.scss');

var _SimpleAutocomplete2 = _interopRequireDefault(_SimpleAutocomplete);

var _SimpleBubble = require('./SimpleBubble.scss');

var _SimpleBubble2 = _interopRequireDefault(_SimpleBubble);

var _SimpleButton = require('./SimpleButton.scss');

var _SimpleButton2 = _interopRequireDefault(_SimpleButton);

var _SimpleCheckbox = require('./SimpleCheckbox.scss');

var _SimpleCheckbox2 = _interopRequireDefault(_SimpleCheckbox);

var _SimpleFlex = require('./SimpleFlex.scss');

var _SimpleFlex2 = _interopRequireDefault(_SimpleFlex);

var _SimpleFormField = require('./SimpleFormField.scss');

var _SimpleFormField2 = _interopRequireDefault(_SimpleFormField);

var _SimpleGrid = require('./SimpleGrid.scss');

var _SimpleGrid2 = _interopRequireDefault(_SimpleGrid);

var _SimpleGutter = require('./SimpleGutter.scss');

var _SimpleGutter2 = _interopRequireDefault(_SimpleGutter);

var _SimpleHeader = require('./SimpleHeader.scss');

var _SimpleHeader2 = _interopRequireDefault(_SimpleHeader);

var _SimpleInfiniteScroll = require('./SimpleInfiniteScroll.scss');

var _SimpleInfiniteScroll2 = _interopRequireDefault(_SimpleInfiniteScroll);

var _SimpleInput = require('./SimpleInput.scss');

var _SimpleInput2 = _interopRequireDefault(_SimpleInput);

var _SimpleLoadingSpinner = require('./SimpleLoadingSpinner.scss');

var _SimpleLoadingSpinner2 = _interopRequireDefault(_SimpleLoadingSpinner);

var _SimpleModal = require('./SimpleModal.scss');

var _SimpleModal2 = _interopRequireDefault(_SimpleModal);

var _SimpleOptions = require('./SimpleOptions.scss');

var _SimpleOptions2 = _interopRequireDefault(_SimpleOptions);

var _SimpleProgressBar = require('./SimpleProgressBar.scss');

var _SimpleProgressBar2 = _interopRequireDefault(_SimpleProgressBar);

var _SimpleRadio = require('./SimpleRadio.scss');

var _SimpleRadio2 = _interopRequireDefault(_SimpleRadio);

var _SimpleSelect = require('./SimpleSelect.scss');

var _SimpleSelect2 = _interopRequireDefault(_SimpleSelect);

var _SimpleSwitch = require('./SimpleSwitch.scss');

var _SimpleSwitch2 = _interopRequireDefault(_SimpleSwitch);

var _SimpleTextArea = require('./SimpleTextArea.scss');

var _SimpleTextArea2 = _interopRequireDefault(_SimpleTextArea);

var _SimpleToggler = require('./SimpleToggler.scss');

var _SimpleToggler2 = _interopRequireDefault(_SimpleToggler);

var _SimpleTooltip = require('./SimpleTooltip.scss');

var _SimpleTooltip2 = _interopRequireDefault(_SimpleTooltip);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
// import theme IDENTIFIERS constants


// css modules plugin converts all imports below into plain objects


// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
var SimpleTheme = exports.SimpleTheme = (_SimpleTheme = {}, _defineProperty(_SimpleTheme, _components.IDENTIFIERS.AUTOCOMPLETE, _SimpleAutocomplete2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUBBLE, _SimpleBubble2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUTTON, _SimpleButton2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.CHECKBOX, _SimpleCheckbox2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FLEX, _SimpleFlex2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FORM_FIELD, _SimpleFormField2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GRID, _SimpleGrid2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GUTTER, _SimpleGutter2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.HEADER, _SimpleHeader2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INFINITE_SCROLL, _SimpleInfiniteScroll2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INPUT, _SimpleInput2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.LOADING_SPINNER, _SimpleLoadingSpinner2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.MODAL, _SimpleModal2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.OPTIONS, _SimpleOptions2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.PROGRESS_BAR, _SimpleProgressBar2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.RADIO, _SimpleRadio2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SELECT, _SimpleSelect2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SWITCH, _SimpleSwitch2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TEXT_AREA, _SimpleTextArea2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOGGLER, _SimpleToggler2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOOLTIP, _SimpleTooltip2.default), _SimpleTheme);
//# sourceMappingURL=index.js.map