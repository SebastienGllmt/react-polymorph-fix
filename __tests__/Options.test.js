import React from 'react';

import { Options } from '../source/components/Options';
import { OptionsSkin } from '../source/skins/simple/OptionsSkin';
import { renderInSimpleTheme } from './helpers/theming';

const MNEMONIC_WORDS = [
  'home',
  'cat',
  'dog',
  'fish',
  'hide',
  'hover',
  'duck',
  'category',
  'join',
  'paper',
  'box',
  'tab'
];

const COUNTRIES_OPTIONS = [
  { german: 'Frankreich', english: 'France' },
  { german: 'Spanien', english: 'Spain' },
  { german: 'Kroatien', english: 'Croatia' },
  { german: 'Vereinigten Staaten', english: 'United States' },
  { german: 'Ã–sterreich', english: 'Austria' }
];

test('Options renders correctly', () => {
  const component = renderInSimpleTheme(
    <Options
      options={MNEMONIC_WORDS}
      skin={OptionsSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Options uses render prop - render', () => {
  const component = renderInSimpleTheme(
    <Options
      options={MNEMONIC_WORDS}
      skin={OptionsSkin}
      render={getOptionProps => {
        const { options } = getOptionProps();
        return options.map((option, index) => (
          <li key={index}>
            <span>{option}</span>
          </li>
        ));
      }}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Options uses render prop - optionRenderer', () => {
  const component = renderInSimpleTheme(
    <Options
      options={COUNTRIES_OPTIONS}
      skin={OptionsSkin}
      optionRenderer={option => (
        <div>
          <span>{option.german}</span>
          <span>{option.english}</span>
        </div>
      )}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
