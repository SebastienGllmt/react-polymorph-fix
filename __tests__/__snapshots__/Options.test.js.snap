// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options renders correctly 1`] = `
<div
  className="options firstOptionHighlighted root isFloating isHidden"
  style={null}
>
  <div
    className="bubble"
    data-bubble-container={true}
  >
    <ul
      className="ul"
    >
      <li
        aria-hidden={true}
        className="option highlightedOption"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        home
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        cat
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        dog
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        fish
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        hide
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        hover
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        duck
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        category
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        join
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        paper
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        box
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        tab
      </li>
    </ul>
  </div>
  <span
    className="arrow"
    data-bubble-arrow={true}
  />
</div>
`;

exports[`Options uses render prop - optionRenderer 1`] = `
<div
  className="options firstOptionHighlighted root isFloating isHidden"
  style={null}
>
  <div
    className="bubble"
    data-bubble-container={true}
  >
    <ul
      className="ul"
    >
      <li
        aria-hidden={true}
        className="option highlightedOption"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        <div>
          <span>
            Frankreich
          </span>
          <span>
            France
          </span>
        </div>
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        <div>
          <span>
            Spanien
          </span>
          <span>
            Spain
          </span>
        </div>
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        <div>
          <span>
            Kroatien
          </span>
          <span>
            Croatia
          </span>
        </div>
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        <div>
          <span>
            Vereinigten Staaten
          </span>
          <span>
            United States
          </span>
        </div>
      </li>
      <li
        aria-hidden={true}
        className="option"
        onClick={[Function]}
        onMouseEnter={[Function]}
        role="presentation"
      >
        <div>
          <span>
            Ã–sterreich
          </span>
          <span>
            Austria
          </span>
        </div>
      </li>
    </ul>
  </div>
  <span
    className="arrow"
    data-bubble-arrow={true}
  />
</div>
`;

exports[`Options uses render prop - render 1`] = `
<div
  className="options firstOptionHighlighted root isFloating isHidden"
  style={null}
>
  <div
    className="bubble"
    data-bubble-container={true}
  >
    <ul
      className="ul"
    >
      <li>
        <span>
          home
        </span>
      </li>
      <li>
        <span>
          cat
        </span>
      </li>
      <li>
        <span>
          dog
        </span>
      </li>
      <li>
        <span>
          fish
        </span>
      </li>
      <li>
        <span>
          hide
        </span>
      </li>
      <li>
        <span>
          hover
        </span>
      </li>
      <li>
        <span>
          duck
        </span>
      </li>
      <li>
        <span>
          category
        </span>
      </li>
      <li>
        <span>
          join
        </span>
      </li>
      <li>
        <span>
          paper
        </span>
      </li>
      <li>
        <span>
          box
        </span>
      </li>
      <li>
        <span>
          tab
        </span>
      </li>
    </ul>
  </div>
  <span
    className="arrow"
    data-bubble-arrow={true}
  />
</div>
`;
